parity-crypto/src/publickey/ecdh.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+//! ECDH scheme functions for key agreement
+
+use secp256k1::{self, ecdh, key};
+use super::{Error, Secret, Public, SECP256K1};
+
+/// Agree on a shared secret
+pub fn agree(secret: &Secret, public: &Public) -> Result<Secret, Error> {
```

Comment:
I think this is a "hot" call. Would be great to have a benchmark for it.

Reply:
Benchmark is added.